src/mui-omni-search.js

```diff
@@ -0,0 +1,161 @@
+import React, { useState, useEffect } from 'react'
+import PropTypes from 'prop-types'
+
+import Autocomplete from '@material-ui/lab/Autocomplete'
+import SearchIcon from '@material-ui/icons/Search'
+import Paper from '@material-ui/core/Paper'
+import InputBase from '@material-ui/core/InputBase'
+import CircularProgress from '@material-ui/core/CircularProgress'
+import { makeStyles } from '@material-ui/core/styles'
+
+import { searchIndexLevelMap } from './constants/campaing-level'
+
+
+const useStyles = makeStyles(() => ({
+  container: { margin: '5px' },
+  paper: {
+    minWidth: '80px',
+    padding: 5,
+    textAlign: 'center'
+  },
+  content: { paddingLeft: 20 },
+  input: {
+    backgroundColor: 'white',
+    borderRadius: '0.285714rem',
+    padding: '6px 15px',
+    width: '30rem'
+  }
+}))
+
+
+/**
+ * OmniSearch Component - overlord SUI parity written with MUI 
+ * @function OmniSearch
+ * @property {function} debouncedSearch - the debounced search function
+ * @property {array} results - the results of the search
+ * @property {function} getSelection - a function that retrieves the details of the selected item (object to be stored in a state)
+ * @property {object} autocompleteProps - override props from <Autocomplete>
+ * @property {object} inputProps - override props from <InputBase>
+ */
+
+export default function OmniSearch({
+  debouncedSearch,
+  results,
+  getSelection,
+  autocompleteProps,
+  inputProps
+}) {
+  const [input, setInput] = useState('')
+  const [value, setValue] = useState(null)
+  const [isOpen, setisOpen] = useState(false)
+  const [loading, setLoading] = useState(false)
+  const [result, setResult] = useState([])
+
+  const classes = useStyles()
+
+  useEffect(() => {
+    if (results.length) {
+      let catogorizedResults = []
+      Object.keys(searchIndexLevelMap).forEach((category) => {
+        let typeResults = results.filter(
+          (result) => result.level === searchIndexLevelMap[category]
+        )
+        typeResults = typeResults.map((c) => ({
+          category: category,
+          name: c.name,
+          id: c.id,
+          level: c.level,
+        }))
+        if (typeResults.length !== 0) {
+          catogorizedResults = [...catogorizedResults, ...typeResults]
+        }
+      })
+      setResult(catogorizedResults)
+      setLoading(false)
+    } else {
+      setLoading(false)
```

Comment:
if this is anyway `setLoading(false)`, rather set it outside of the if/else flow after the `if` block